### PR TITLE
Introduce admin console

### PR DESCRIPTION
**Pull request by Saphs**

### CODE REVIEW

File: src/app/components/admin-console/admin-console.component.html
Comment (DanielLiebler) [inline src/app/components/admin-console/admin-console.component.html:1]: where does this xmlns come from? I dont think this should belong here

File: src/app/components/admin-console/admin-console.component.html [hunk 1]
```diff
@@ -0,0 +1,200 @@
+<div class="container-fluid my-5" xmlns="http://www.w3.org/1999/html">
```
Comment (DanielLiebler) [inline src/app/components/admin-console/admin-console.component.html:1]: https://www.youtube.com/watch?v=xTKXnfHByX8

File: src/app/services/object-loader/asset-loader.ts
Comment (DanielLiebler) [inline src/app/services/object-loader/asset-loader.ts:1]: why not static class and static functions? should be static and then you dont need to instantiate the Assetloader.

File: src/app/services/object-loader/object-loader.service.ts [hunk 2]
```diff
@@ -13,294 +12,69 @@ import {
...
-      path: '/assets/cubemaps/bridge-skyboxes/Bridge2/',
-      px: 'posx.jpg',
-      py: 'posy.jpg',
-      pz: 'posz.jpg',
-      nx: 'negx.jpg',
-      ny: 'negy.jpg',
-      nz: 'negz.jpg',
-    },
-  ];
+  private al: AssetLoader = new AssetLoader();
```
Comment (DanielLiebler) [inline src/app/services/object-loader/object-loader.service.ts:30]: See above notice about static classes

File: src/app/services/object-loader/loaderTypes.ts [hunk 3]
```diff
@@ -61,4 +72,33 @@ export interface CubeMap {
...
+  ny: string;
+  nz: string;
+  path: string;
+  px: string;
+  py: string;
+  pz: string;
+  tex: CubeTexture;
+}
+
+export interface Prop<T> {
```
Comment (DanielLiebler) [inline src/app/services/object-loader/loaderTypes.ts:99]: if not used, please remove

File: src/app/services/object-loader/object-loader.service.ts [hunk 4]
```diff
@@ -479,36 +231,25 @@ export class ObjectLoaderService {
...
 
   loadGameTileTexture(texUrl: string): Observable<Texture> {
+    const defaultTileTexture: Texture = this.al.loadTexture(this.al.defaultTileTexturePath);
+    this.loadedObjects.push(defaultTileTexture);
     return new Observable<Texture>((observer: Observer<Texture>) => {
-      this.tLoader.load(
-        texUrl,
-        (texture) => {
+      const t: Texture = this.al.loadTexture(
+        'http://localhost:25670/api/assets/static/asset_1698709289146.png',
```
Comment (DanielLiebler) [inline src/app/services/object-loader/object-loader.service.ts:251]: :(

File: src/app/services/object-loader/object-loader.service.ts [hunk 5]
```diff
@@ -683,34 +402,44 @@ export class ObjectLoaderService {
...
+    ctx.lineTo(x + r, y + h);
+    ctx.quadraticCurveTo(x, y + h, x, y + h - r);
+    ctx.lineTo(x, y + r);
+    ctx.quadraticCurveTo(x, y, x + r, y);
+    ctx.closePath();
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  dispose(): void {
```
Comment (DanielLiebler) [inline src/app/services/object-loader/object-loader.service.ts:430]: delete unused

File: src/app/services/object-loader/asset-loader.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+  /** Helper functions to make loader access cleaner */
+  public loadTexture(
+    path: string,
+    onLoad?: (texture: Texture) => void,
+    onProgress?: (event: ProgressEvent) => void,
+    onError?: (event: ErrorEvent) => void
+  ): Texture {
+    console.log('Loading texture: ', path);
+    const default_onLoad = (t: Texture) => {
```
Comment (DanielLiebler) [inline src/app/services/object-loader/asset-loader.ts:95]: call supplied functions functions inside the default functions. Then, the default actions will also be executed when a function is supplied

File: src/app/components/admin-console/admin-console.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (Saphs) [reply src/app/components/admin-console/admin-console.component.html:1]: Done

File: src/app/components/game/viewport/viewport.component.ts [hunk 7]
```diff
@@ -50,6 +50,13 @@ export class ViewportComponent implements AfterViewInit, OnDestroy {
 
   ngOnDestroy(): void {
     this.userInteractionController.onDestroy();
+    console.log(this.renderer.info);
```
[the remark this answers is not in the document]
Comment (Saphs) [reply src/app/components/game/viewport/viewport.component.ts:53]: done

File: src/app/services/object-loader/asset-loader.ts
Comment (Saphs) [reply src/app/services/object-loader/asset-loader.ts:1]: it is now

File: src/app/services/object-loader/asset-loader.ts (hunk 6 above)
Comment (Saphs) [reply src/app/services/object-loader/asset-loader.ts:95]: it was done

File: src/app/services/object-loader/loaderTypes.ts (hunk 3 above)
Comment (Saphs) [reply src/app/services/object-loader/loaderTypes.ts:99]: i did

File: src/app/components/game/viewport/viewport.component.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (DanielLiebler) [reply src/app/components/game/viewport/viewport.component.ts:54]: as mentioned before, (aka. read my comments) make console logs more descriptive

File: src/app/services/object-loader/loaderTypes.ts (hunk 3 above)
Comment (DanielLiebler) [reply src/app/services/object-loader/loaderTypes.ts:99]: check again. there are no changes in loaderTypes.ts so far

**Review by DanielLiebler — APPROVED** [review]

**Pull request merged**